src/loader-components/_ExampleTemplate.jsx

```diff
@@ -30,93 +31,68 @@ const defaultProps = {
-  static renderHeader(title) {
-    if (title) {
-      return (
-        <div className={cx('header')}>
-          <h2 className={cx('title')}>
-            {title}
-          </h2>
-        </div>
-      );
-    }
-    return null;
-  }
-
-  static renderDescription(description) {
-    if (description) {
-      return (
-        <div className={cx('description')}>
-          {description}
-        </div>
-      );
-    }
-    return null;
-  }
-
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      isExpanded: props.isExpanded,
-      isBackgroundTransparent: false,
-    };
-
-    this.handleBgToggle = this.handleBgToggle.bind(this);
-    this.handleCodeToggle = this.handleCodeToggle.bind(this);
-  }
-
-  handleBgToggle() {
-    this.setState(prevState => ({ isBackgroundTransparent: !prevState.isBackgroundTransparent }));
-  }
-
-  handleCodeToggle() {
-    this.setState(prevState => ({ isExpanded: !prevState.isExpanded }));
+const renderHeader = (title) => {
+  if (title) {
+    return (
+      <div className={cx('header')}>
+        <h2 className={cx('title')}>
+          {title}
+        </h2>
+      </div>
+    );
   }
+  return null;
+};
 
-  render() {
-    const {
-      example,
-      exampleSrc,
-      title,
-      description,
-    } = this.props;
-
-    const { isExpanded, isBackgroundTransparent } = this.state;
-
+const renderDescription = (description) => {
+  if (description) {
     return (
-      <div className={cx('template')}>
-        {ExampleTemplate.renderHeader(title)}
-        <div className={cx('content', { 'dynamic-content': isBackgroundTransparent })}>
-          {ExampleTemplate.renderDescription(description)}
-          {example}
-        </div>
-        {exampleSrc
-          && (
-          <div className={cx('footer')}>
-            <div className={cx('button-container')}>
-              <button type="button" className={cx('bg-toggle')} onClick={this.handleBgToggle}>
-                Toggle Background
-              </button>
-              <button type="button" className={cx('code-toggle')} onClick={this.handleCodeToggle}>
-                <span className={cx('chevron-left')} />
-                <span>Code</span>
-                <span className={cx('chevron-right')} />
-              </button>
-            </div>
-            {isExpanded
-              && (
-              <div className={cx('code')}>
-                {exampleSrc}
-              </div>
-              )}
-          </div>
-          )}
+      <div className={cx('description')}>
+        {description}
       </div>
     );
   }
-}
+  return null;
+};
+
+const ExampleTemplate = ({
+  example, exampleSrc, title, description, isExpanded,
+}) => {
+  const [codeIsVisible, setCodeIsVisible] = useState(isExpanded);
+  const [isBackgroundTransparent, setIsBackgroundTransparent] = useState(false);
+  const theme = React.useContext(ThemeContext);
+
+  return (
+    <div className={cx('template')}>
```

Comment:
just put the theme class name here once and let it flow through the component
```suggestion
    <div className={cx('template', theme.className)}>
```